Blog search functionality

andipavllo opened the pull request:
@argelius when you have time, please check this.

argelius commented:
Thanks @andipavllo ! I will look at it tomorrow

argelius commented:
@andipavllo Sorry for the delay. I think it would be better if the mouse pointer changes when hovering the magnifying glass.

argelius commented on src/files/scripts/blog_search.js:
```diff
@@ -0,0 +1,63 @@
+    searchIcon.removeEventListener("click", performSearch);
+    searchIcon.addEventListener("click", showSearchBar);
+  }
+})();
```
Missing newline

argelius commented:
One more thing. Does checking for keyCode 13 work on mobile browsers? Maybe it's better to wrap in a form and listen for `submit` event.

argelius commented on src/files/scripts/blog_search.js:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  function performSearch(e) {
+    if (searchBar.value != "" && (e.keyCode==13 || e.target == searchIcon)) {
```
Please use strict comparison

argelius commented on src/files/scripts/blog_search.js:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  function checkOuterClick(e) {
+    if (e.target != searchBar) {
```
Strict comparison

argelius commented on src/files/scripts/blog_search.js:
```diff
@@ -0,0 +1,63 @@
+      searchBar.addEventListener("keypress", performSearch);
+      searchIcon.removeEventListener("click", showSearchBar);
+      searchIcon.addEventListener("click", performSearch);
+    }, 100);
```
What is this delay for?

andipavllo commented:
@argelius thank you for the feedback! I'll fix it the next couple of days.

argelius commented:
Thanks!

andipavllo commented:
@argelius I fixed the code, just two things:

* The search functionality will not be visible on mobile browsers, so we should not be worried about `keyCode 13`
* I was using the setTimeout because the event was being captured by another eventListener, causing undesired behavior. Fixed it by stopping the event propagation and removing the timeout.

argelius commented:
Thanks @andipavllo !

One more small thing. I think there should be a hover effect just like the other links on the toolbar.

andipavllo commented:
@argelius added, merge it if it's ok :)

argelius commented:
Thanks @andipavllo !
